### PR TITLE
Add support for formulas configuration in spacecmd group backup

### PR DESCRIPTION
**Pull request by aaannz**
## What does this PR change?

`spacecmd group_backup` command backups system groups. This backup backups only name and description.  However since some time we allow formulas to be configured for groups as well. This PR enhanced backup (and restore via `spacecmd group_restore`) to include formulas configuration in the group backup.

Previously group backups were simple text files containing description string of the group. For formulas I changed that format to be in a JSON format having special field for description and formula configuration.
Restore procedure now tries to decode JSON to dics and if that fails it assumes that backup is in old format and loads it as a plain file.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- Unit tests were added

- [X] **DONE**

## Links

Fixes #https://github.com/SUSE/spacewalk/issues/15872

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by agraul — CHANGES_REQUESTED** [review]
Thank you for taking care of the fromula confg backup/restore in `spacecmd`! I like the approach you took and think this looks pretty good overall.

There are two small things I'd like to discuss before we merge this, the first is the rpm changelog entry. As I wrote in-line, I think it could be a bit clearer for users. I looked at it before going through the code changes and at the time I didn't really understand the message. The second thing is the usage of `re` for checking the user input.

The rest of my comments are nitpicks. Feel free to ignore them!

File: spacecmd/spacecmd.changes [hunk 1]
```diff
@@ -1,3 +1,4 @@
+- Group backup now backups formulas configuration as well (bsc#1190462)
```
[the remark this answers is not in the document]
Comment (aaannz) [reply spacecmd/spacecmd.changes]: Reworded and extended. 👍🏾

File: spacecmd/src/spacecmd/group.py [hunk 2]
```diff
@@ -355,36 +364,68 @@ def do_group_restore(self, args):
...
-                self.client.systemgroup.update(self.session, groupname, details)
+                logging.debug("Already have %s but the data have changed" % groupname)
+
+                if is_interactive(options):
+                    if current[groupname]['description'] != backup['description']:
+                        print(_("Changing description from:"))
+                        print("\n\"%s\"\nto\n\"%s\"\n" % (current[groupname]['description'], backup['description']))
+                        userinput = prompt_user(_('Continue [y/N]:'))
+
+                        if re.match('y', userinput, re.I):
```
[the remark this answers is not in the document]
Comment (aaannz) [reply spacecmd/src/spacecmd/group.py]: Not really, I blindly copypasted what was used before. Changed.

**Comment by aaannz** [comment]
@agraul will you have time to go over this again? I did the changes based on the review some time back.

**Review by agraul — APPROVED** [review]
Sorry for taking so long, I lost track of this! Thanks for taking the time to implement all those little changes.

**Pull request merged**